### PR TITLE
Implemented: option for logging inventory variance (#85zrx66zh)

### LINKED ISSUE #166: Allow users to enter inventory variances directly from the Update Inventory Count page.
## What is the motivation for adding/enhancing this feature?
Users can currently only update a product's inventory count by entering the exact units available at the store which replaces the inventory count in OMS. Users may be required to log only variance to input the quantity change for a variety of reasons, including damaged units, torn units, lost or found units.

## What are the acceptance criteria?
Users can communicate the quantity that is damaged and not for sale by entering the units that are no longer available.
Only users with FULFILL_INVCUNT_ADMIN or INV_COUNT_ADMIN can use this feature

## Additional information
Count

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #166 

### Screenshots of Visual Changes before/after (If There Are Any)

[Screencast from 20-04-23 06:51:34 PM IST.webm](https://user-images.githubusercontent.com/59442907/233380741-9671dd3c-0270-465d-b87d-37cdfac7dd19.webm)

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vsf-capybara/blob/master/CONTRIBUTING.md)
- [ ] I have changed the target to hacktoberfest branch

### CODE REVIEW

File: src/store/modules/product/actions.ts [hunk 1]
```diff
@@ -84,6 +84,35 @@ const actions: ActionTree<ProductState, RootState> = {
     commit(types.PRODUCT_ADD_TO_UPLD_PRDTS, { product: payload })
   },
 
+  async updateVarianceCount({ commit }, payload) {
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:87]: If we are not doing anything related to state, we should make it an action

File: src/views/count.vue [hunk 2]
```diff
@@ -193,19 +240,57 @@
...
-      }
+      },
+      updateVarianceReason(event: any) {
+        this.product.reason = event.detail.value
+      },
+      async presentAlertOnVarianceUpdate() {
+        if (this.product.variance && this.product.reason) {
+          const alert = await alertController.create({
+            header: this.$t("Log variance"),
+            message: this.$t("Are you sure you want to update the inventory variance?"),
```
Comment (adityasharma7) [inline src/views/count.vue:251]: We should add another line to it like The inventory will be immediately updated and cannot be undone.

File: src/store/modules/util/actions.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { UtilService } from '@/services/UtilService'
+import { ActionTree } from 'vuex'
+import RootState from '@/store/RootState'
+import UserState from './UtilState'
+import * as types from './mutation-types'
+import { hasError } from '@/utils'
+
+const actions: ActionTree<UserState, RootState> = {
+  async fetchVarianceReasons({ commit }) {
```
Comment (adityasharma7) [inline src/store/modules/util/actions.ts:9]: Why to use state when we use it in single page?

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/util/actions.ts (hunk 3 above)
Comment (k2maan) [reply src/store/modules/util/actions.ts]: Updated sir

File: src/store/modules/product/actions.ts (hunk 1 above)
Comment (k2maan) [reply src/store/modules/product/actions.ts]: Made it a method

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**